### PR TITLE
feat: allow filter on fill/order source for sales ws

### PR DESCRIPTION
**Pull request by tv3636**

### CODE REVIEW

File: packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-job.ts [hunk 1]
```diff
@@ -143,6 +143,8 @@ export class SaleWebsocketEventsTriggerQueueJob extends AbstractRabbitMqJobHandl
           contract: data.after.contract,
           maker: data.after.maker,
           taker: data.after.taker,
+          fillSource: result.fillSource ?? "",
```
Comment (nofir) [inline packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-job.ts:146]: Isnt the fillSource / orderSource the whole obj? it should be based on the domain

**Review by nofir — CHANGES_REQUESTED** [review]

File: packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-job.ts (hunk 1 above)
Comment (tv3636) [reply packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-job.ts:146]: `getJoiSaleObject` returns just the domain: https://github.com/reservoirprotocol/indexer/blob/8dd3a14d4e16bdd0fb254e48b17b23ad928decae/packages/indexer/src/common/joi.ts#L909

Comment (nofir) [reply packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-job.ts:146]: i missed that, ok makes sense. any reason to set it to ""? just trying to understand if not including it (undefined), creates an issues?

Comment (tv3636) [reply packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-job.ts:146]: the ws publisher expects all tags to be `string`, alternatively we could allow the tags to be null and remove the empty string here

Comment (nofir) [reply packages/indexer/src/jobs/websocket-events/sale-websocket-events-trigger-job.ts:146]: and again, if the firehose, just ignores empty strings, its probably ok to keep like that, i just thought it was cleaner

**Review by nofir — APPROVED** [review]

**Pull request merged**